pkg/cache/snapshot_test.go

```diff
@@ -1331,3 +1204,166 @@ func TestSnapshotAddRemoveWorkloadWithLendingLimit(t *testing.T) {
 		})
 	}
 }
+
+func TestResourceGroups(t *testing.T) {
```

Comment:
I disagree with these changes... We are replacing a simple diff with more test code.

It's ok if the test code is repetitive (https://enterprisecraftsmanship.com/posts/dry-damp-unit-tests/)

If ResourceGroups are not relevant for a certain test, we can simply remove them from the input data.